### PR TITLE
Migrate to new gRPC library by ASP.NET Core 3

### PR DESCRIPTION
**Pull request by ankitkumarr**
This is building on top of the other refactor PR. I set the base to the branch for that PR just to have separate diffs.

Once that's merged I will update the base here.

Fixes https://github.com/Azure/azure-functions-host/issues/5449

### CODE REVIEW

File: src/WebJobs.Script.Grpc/WebJobs.Script.Grpc.csproj [hunk 1]
```diff
@@ -20,12 +21,12 @@
...
   <ItemGroup>
-    <PackageReference Include="Google.Protobuf" Version="3.7.0" />
+    <PackageReference Include="Grpc.AspNetCore" Version="2.26.0" />
     <PackageReference Include="Google.Protobuf.Tools" Version="3.7.0" />
-    <PackageReference Include="Grpc.Core" Version="1.20.1" />
-    <PackageReference Include="Grpc.Tools" Version="1.20.1">
+    <PackageReference Include="Grpc.Tools" Version="2.25.0" >
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
+
```
Comment (ankitkumarr) [inline src/WebJobs.Script.Grpc/WebJobs.Script.Grpc.csproj]: will fix accidental empty lines

File: src/WebJobs.Script.Grpc/Server/GrpcServer.cs [hunk 2]
```diff
@@ -2,51 +2,48 @@
...
 
         protected virtual void Dispose(bool disposing)
         {
             if (!_disposed)
             {
                 if (disposing)
                 {
-                    _server.ShutdownAsync();
+                    _grpcHost.StopAsync().GetAwaiter().GetResult();
                 }
```
Comment (ankitkumarr) [inline src/WebJobs.Script.Grpc/Server/GrpcServer.cs]: Should probably also dispose the grpcHost

File: src/WebJobs.Script/Workers/WorkerHelpers.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Net;
+using System.Net.Sockets;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers
+{
+    public static class WorkerHelpers
```
Comment (pragnagopa) [inline src/WebJobs.Script/Workers/WorkerHelpers.cs]: Nit: rename WorkerHelpers to WorkerUtilities to follow similar naming convention as existing helper classes

File: src/WebJobs.Script.Grpc/GrpcHostBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+using System.Text;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Azure.WebJobs.Script.Eventing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc
+{
+    internal static class GrpcHostBuilder
```
[the remark this answers is not in the document]
Comment (ankitkumarr) [reply src/WebJobs.Script.Grpc/GrpcHostBuilder.cs]: It is already in the Grpc project :)

[the remark this answers is not in the document]
Comment (pragnagopa) [reply src/WebJobs.Script.Grpc/GrpcHostBuilder.cs]: my bad. Misread :)

**Comment by pragnagopa** [comment]
@ankitkumarr / @fabiocav  - So we plan to get these changes in?

**Comment by ankitkumarr** [comment]
@pragnagopa, I need to add this to one of the next Sprints. I can probably pick it up in the next Sprint. I think I need to rebase and update some things here as they went pretty out of date.

**Comment by pragnagopa** [comment]
As this is a big change, @ankitkumarr  - when you have changes ready please work with @yojagad to figure out if it is possible to kick off a perf run before merging the changes. Thanks.

**Comment by fabiocav** [comment]
Closing stale PRs. Please reopen if still relevant.

**Pull request closed without merging**